### ISSUE TITLE
Linux support

### ISSUE THREAD

**Issue by Schmavery** [Schmavery/reprocessing#63]
Unfortunately reasongl doesn't seem to be building on linux. Happy to help anyone who wants to try and figure out a solution to this, otherwise, hope I do soon.
https://github.com/bsansouci/reasongl/issues/2

**Comment by Schmavery** [comment]
This is basically done, but audio is tricky on linux so we want to tweak the audio support so that it prints a warning rather than crashing if an audio device can't be found.

**Comment by zploskey** [comment]
Happy to help out debugging Linux issues. I see the crash when it can't find an audio device. Wondering what the issue with finding the audio device is. I installed alsa and pulseaudio development libraries to get things to build.

**Comment by Schmavery** [comment]
@zploskey this is the file that would need to be modified to avoid the crash on 0 audio devices:
https://github.com/bsansouci/tsdl/blob/master/src/tsdl_new.c

IIRC, the part that fails is here: https://github.com/bsansouci/tsdl/blob/421f63717bb7bb2d7dbb8ac54706a86ffb89f862/src/tsdl_new.c#L126-L144

I can't for the life of me remember why we would call SDL_OpenAudioDevice twice.. that might be a copy/paste bug but not sure.

Basically we would want to just warn in that case, and set some flag so that in the rest of the file, we can disable any audio-related code from running.

Does that help? Happy to fill in anything else that's confusing.

**Comment by bsansouci** [comment]
@Schmavery yeah the first one might be wrong? Seems like we should be filling up `device->want` first and then get a device... But I'm blind here lol

**Comment by Schmavery** [comment]
To be honest, it's totally possible (read: pretty likely) that we're handing frequency/samples setting completely wrong.. I hacked audio support together in like 2 days before a game jam so that we could have sounds in our game 😆

**Comment by zploskey** [comment]
I guess removing the first if with the call to `SDL_OpenAudioDevice` would at least allow it to continue without finding an audio device. Then it's a matter of figuring out why it's not finding one.

**Comment by Schmavery** [comment]
Will close this in favour of the more specific problem described in #74

**Comment by jdeisenberg** [comment]
This is an FYI based on discussion with Schmavery on 12 April. On Fedora 29, you have to install these (listed as pairs of library and development information)
```
mesa mesa-libGL-devel
SDL2 SDL2-devel
pusleaudio-libs pulseaudio-libs-devel
alsa-libs alsa-libs-devel
ibus ibus-devel
fcitx fcitx-devel
```
